src/Xamarin.Android.Tools.ApiXmlAdjuster/JavaApi.XmlModel.cs

```diff
@@ -24,15 +26,61 @@ public JavaPackage (JavaApi? parent)
 		{
 			Parent = parent;
 			
-			Types = new List<JavaType> ();
+			Types = new Dictionary<string, List<JavaType>> (StringComparer.OrdinalIgnoreCase);
 		}
 		
 		public  JavaApi?                Parent              { get; private set; }
 
 		public  string?                 Name                { get; set; }
 		public  string?                 JniName             { get; set; }
-		public  IList<JavaType>         Types               { get; set; }
-		
+
+		// Yes, there can be multiple types with the same *Java* name.
+		// For example:
+		// - MyInterface
+		// - MyInterfaceConsts
+		// It's debatable whether we handle this "properly", as most callers just
+		// do `First ()`, but it's been working for years so I'm not changing it.
+		// Exposes an IReadOnlyDictionary so caller cannot bypass our AddType/RemoveType code.
+		public  IReadOnlyDictionary<string, List<JavaType>>   Types               { get; }
+
+		// Use this for a flat list of *all* types
+		public IEnumerable<JavaType> AllTypes => Types.Values.SelectMany (v => v);
+
+		public void AddType (string name, JavaType type)
+		{
+			// If this is a duplicate key, add it to existing list
+			if (Types.TryGetValue (name, out var list)) {
+				list.Add (type);
+				return;
+			}
+
+			// Add to a new list
+			var new_list = new List<JavaType> ();
+			new_list.Add (type);
+
+			((IDictionary<string, List<JavaType>>)Types).Add (name, new_list);
```

Comment:
Instead of these casts, would it make more sense to have a `Dictionary<string, List<JavaType>> types` *field*, which the `Types` property returns, but allows "direct" access w/o these casts?

```csharp
public  IReadOnlyDictionary<string, List<JavaType>>   Types               => types;
```

Reply:
Changed.